applications/FluidDynamicsApplication/python_scripts/fluid_solver.py

```diff
@@ -259,3 +273,162 @@ def _SetNodalProperties(self):
+
+    def get_builder_and_solver(self):
+        if not hasattr(self, '_builder_and_solver'):
+            self._builder_and_solver = self._create_builder_and_solver()
+        return self._builder_and_solver
+
+    def get_solution_strategy(self):
+        if not hasattr(self, '_solution_strategy'):
+            self._solution_strategy = self._create_solution_strategy()
+        return self._solution_strategy
+
+    def _create_estimate_dt_utility(self):
+        domain_size = self.GetComputingModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        if domain_size == 2:
+            estimate_dt_utility = KratosCFD.EstimateDtUtility2D(
+                self.GetComputingModelPart(),
+                self.settings["time_stepping"])
+        else:
+            estimate_dt_utility = KratosCFD.EstimateDtUtility3D(
+                self.GetComputingModelPart(),
+                self.settings["time_stepping"])
+
+        return estimate_dt_utility
+
+    def _create_solution_scheme(self):
+        domain_size = self.GetComputingModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        # Cases in which the element manages the time integration
+        if self.element_integrates_in_time:
+            # "Fake" scheme for those cases in where the element manages the time integration
+            # It is required to perform the nodal update once the current time step is solved
+            solution_scheme = KratosMultiphysics.ResidualBasedIncrementalUpdateStaticSchemeSlip(
+                domain_size,
+                domain_size + 1)
+            # In case the BDF2 scheme is used inside the element, the BDF time discretization utility is required to update the BDF coefficients
+            if (self.settings["time_scheme"].GetString() == "bdf2"):
+                time_order = 2
+                self.time_discretization = KratosMultiphysics.TimeDiscretization.BDF(time_order)
+            else:
+                err_msg = "Requested elemental time scheme \"" + self.settings["time_scheme"].GetString()+ "\" is not available.\n"
+                err_msg += "Available options are: \"bdf2\""
+                raise Exception(err_msg)
+        # Cases in which a time scheme manages the time integration
+        else:
+            # Time scheme without turbulence modelling
+            if not hasattr(self, "_turbulence_model_solver"):
+                # Bossak time integration scheme
+                if self.settings["time_scheme"].GetString() == "bossak":
+                    if self.settings["consider_periodic_conditions"].GetBool() == True:
+                        solution_scheme = KratosCFD.ResidualBasedPredictorCorrectorVelocityBossakSchemeTurbulent(
+                            self.settings["alpha"].GetDouble(),
+                            domain_size,
+                            KratosCFD.PATCH_INDEX)
+                    else:
+                        solution_scheme = KratosCFD.ResidualBasedPredictorCorrectorVelocityBossakSchemeTurbulent(
```

Comment:
this is `time_scheme` not `solution_scheme`

Reply:
I followed the naming convention in the `StructuralMechanicsApplication`. Consider that there are cases in which we use a fake scheme just to do the solution update or steady cases in which the concept of time is meaningless.

Reply:
Then what about the case of dynamic simulations, which most of the fluid cases/simulations are !

Reply:
IMO what StructuralMechanicsApp does should also be changed to a neutral name `scheme`

Reply:
I would also use the same name as in C++

The code in the Structural has been there for a while, I wouldn't necessarily follow the naming there

Reply:
:+1: I'll rename all to `scheme`.